### ISSUE TITLE
Event notifications

### ISSUE THREAD

**Issue by marzavec** [hack-chat/hackchat-client#17]
Notifications need to be added. I was thinking of using [react-web-notification](https://www.npmjs.com/package/react-web-notification) for compatibility reasons and it's built-in service worker handling.

Notifications should be built to check if the window has focus, as well as what channel currently has focus, allowing mentions in other background channels to be displayed.

**Comment by KnowsCount** [comment]
sounds should play a big role in the notification part, concluding from my past experience using hack.chat. some sort of clearer reminder, at least. but as stated in [react-web-notification](https://www.npmjs.com/package/react-web-notification):

> Notification.sound Notification.sound is not supported in any browser. You can emulate it with onShow callback. see example.

there might be issues with sound... can this be solved?—I am personally not quite familiar with this notification thingy.

**Comment by marzavec** [comment]
> can this be solved?

Yep, should be easy enough, compatibility will be the biggest hurdle though.

**Comment by marzavec** [comment]
Completed with commit c791ee9848638eca44983215179823c90f43cbc5
